### ISSUE TITLE
[apex]

### ISSUE THREAD

**Issue by brandbarr** [pmd/pmd#5424]
**Affects PMD Version:**  7.8.0 and 7.9.0

**Rule:**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd/pmd_rules_apex_bestpractices.html#unusedlocalvariable>

**Description:**
When variable is used in a string as a [bind](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_SOQL_variables.htm), `UnusedLocalVarable` is incorrectly flagged as an issue. 

**Code Sample demonstrating the issue:**

```java
  public void List<String> exampleFalsePositiveMethod() {
    Set<String> testStatuses = RequestsConstants.TEST_FILTER_STRING;

    List<String> wheres = new List<String>();
    wheres.add('Status__c IN :testStatuses');
    
    return wheres;
  }
```

Output:
```
    rule: UnusedLocalVariable
    msg:  Variable 'testStatuses' defined but not used
    code: Set<String> testStatuses = RequestsConstants.TEST_FILTER_STRING;
```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** CLI
